algolia/algoliasearch-sfcc-b2c
# feat(insights): new addToCart event format

Algolia Insights library now contains dedicated `addToCart` events that accept more information about the products added to the cart: https://www.algolia.com/doc/api-reference/api-methods/added-to-cart-object-ids-after-search/

We were hooking the previous conversion events to the `click` events of the `add-to-cart` buttons, but SFRA triggers dedicated events that send the information we need (c.f. [`addToCart` function](https://github.com/SalesforceCommerceCloud/storefront-reference-architecture/blob/1d7d4d987d681a11b045746618aec744b4409540/cartridges/app_storefront_base/cartridge/client/default/js/product/base.js#L624)):

- A `updateAddToCartFormData` event triggered before the addToCart call and containing:
  ```js
  {
      pid: "productId", // defined in case of a single product added
      pidsObj: [ { pid: "productId", qty: "1" } ], // defined in case of a product set
      quantity: "1" // accurate in case of a single product
  }
  ```
- A `product:afterAddToCart` triggered after the addToCart call and containing the current `cart` with all products and their information (id, list price, sales price, ...). With the information stored in the `updateAddToCartFormData` event, we can compute and send the Algolia insights `addToCart` event for the newly added products.

### Changes

- Reintroduced the `search-insights` library. The one shipped with InstantSearch and Autocomplete is in version `2.6.0` and doesn't contain the new events yet.
- Reimplemented the Algolia insights events based on SFRA events
- Pass the `productIndex` at initialization of the insights library, in case users open a product page directly (`index` is a required parameter)
- Bonus: Added a `postinstall` step that copy the Algolia frontend libraries systematically after a `npm install`

### How to test

Add products to your cart and inspect the network requests. You should see insights calls with:
- the v2.11.0 of the Insights library
- the new properties in the payload: `eventSubtype: "addToCart"` and `objectData`

---
SFCC-197
